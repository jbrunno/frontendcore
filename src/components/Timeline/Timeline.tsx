import React, { Ref } from 'react';
import {
  Timeline as MuiTimeline,
  TimelineProps as MuiTimelineProps,
} from '@mui/lab';

export type TimelineProps = MuiTimelineProps & {
  ref?: Ref<HTMLUListElement>;
};

export function Timeline(props: TimelineProps) {
  return <MuiTimeline {...props} />;
}

export default Timeline;
