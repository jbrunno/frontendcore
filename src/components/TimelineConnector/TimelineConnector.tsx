import React from 'react';
import {
  TimelineConnector as MuiTimelineConnector,
  TimelineConnectorProps as MuiTimelineConnectorProps,
} from '@mui/lab';

export type TimelineConnectorProps = MuiTimelineConnectorProps;

export function TimelineConnector(props: TimelineConnectorProps) {
  return <MuiTimelineConnector {...props} />;
}

export default TimelineConnector;
