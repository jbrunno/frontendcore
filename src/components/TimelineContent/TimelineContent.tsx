import React from 'react';
import {
  TimelineContent as MuiTimelineContent,
  TimelineContentProps as MuiTimelineContentProps,
} from '@mui/lab';

export type TimelineContentProps = MuiTimelineContentProps;

export function TimelineContent(props: TimelineContentProps) {
  return <MuiTimelineContent {...props} />;
}

export default TimelineContent;
