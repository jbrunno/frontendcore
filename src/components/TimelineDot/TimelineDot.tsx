import React from 'react';
import {
  TimelineDot as MuiTimelineDot,
  TimelineDotProps as MuiTimelineDotProps,
} from '@mui/lab';

export type TimelineDotProps = MuiTimelineDotProps;

export function TimelineDot(props: TimelineDotProps) {
  return <MuiTimelineDot {...props} />;
}

export default TimelineDot;
