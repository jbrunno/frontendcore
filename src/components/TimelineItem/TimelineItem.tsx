import React from 'react';
import {
  TimelineItem as MuiTimelineItem,
  TimelineItemProps as MuiTimelineItemProps,
} from '@mui/lab';

export type TimelineItemProps = MuiTimelineItemProps;

export function TimelineItem(props: TimelineItemProps) {
  return <MuiTimelineItem {...props} />;
}

export default TimelineItem;
