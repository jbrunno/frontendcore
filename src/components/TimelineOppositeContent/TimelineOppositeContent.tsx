import React from 'react';
import {
  TimelineOppositeContent as MuiTimelineOppositeContent,
  TimelineOppositeContentProps as MuiTimelineOppositeContentProps,
} from '@mui/lab';

export type TimelineOppositeContentProps = MuiTimelineOppositeContentProps;

export function TimelineOppositeContent(props: TimelineOppositeContentProps) {
  return <MuiTimelineOppositeContent {...props} />;
}

export default TimelineOppositeContent;
