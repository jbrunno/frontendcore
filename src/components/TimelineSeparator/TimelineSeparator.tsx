import React from 'react';
import {
  TimelineSeparator as MuiTimelineSeparator,
  TimelineSeparatorProps as MuiTimelineSeparatorProps,
} from '@mui/lab';

export type TimelineSeparatorProps = MuiTimelineSeparatorProps;

export function TimelineSeparator(props: TimelineSeparatorProps) {
  return <MuiTimelineSeparator {...props} />;
}

export default TimelineSeparator;
