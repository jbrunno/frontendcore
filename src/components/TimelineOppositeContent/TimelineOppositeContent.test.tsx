import React from 'react';
import { render } from '@test-utils/component-testing';
import Timeline from '../Timeline/Timeline';
import TimelineItem from '../TimelineItem/TimelineItem';
import TimelineSeparator from '../TimelineSeparator/TimelineSeparator';
import TimelineDot from '../TimelineDot/TimelineDot';
import TimelineConnector from '../TimelineConnector/TimelineConnector';
import TimelineContent from '../TimelineContent/TimelineContent';
import TimelineOppositeContent from './TimelineOppositeContent';

test('renders the TimelineConnector', () => {
  const { container } = render(
    <Timeline position="left">
      <TimelineItem>
        <TimelineOppositeContent>Opposite</TimelineOppositeContent>
        <TimelineSeparator>
          <TimelineDot />
          <TimelineConnector />
        </TimelineSeparator>
        <TimelineContent>Test</TimelineContent>
      </TimelineItem>
      <TimelineItem>
        <TimelineOppositeContent>Opposite</TimelineOppositeContent>
        <TimelineSeparator>
          <TimelineDot />
          <TimelineConnector />
        </TimelineSeparator>
        <TimelineContent>Test</TimelineContent>
      </TimelineItem>
      <TimelineItem>
        <TimelineOppositeContent>Opposite</TimelineOppositeContent>
        <TimelineSeparator>
          <TimelineDot />
        </TimelineSeparator>
        <TimelineContent>Test</TimelineContent>
      </TimelineItem>
    </Timeline>,
  );
  expect(container).toMatchSnapshot();
});
