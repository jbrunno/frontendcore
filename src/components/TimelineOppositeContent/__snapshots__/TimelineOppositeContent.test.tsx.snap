// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TimelineConnector 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 6px 16px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-1 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  min-height: 70px;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.emotion-1:before {
  content: "";
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 6px 16px;
}

.emotion-2 {
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  padding: 6px 16px;
  margin-right: auto;
  text-align: left;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  border-style: solid;
  border-width: 2px;
  padding: 4px;
  border-radius: 50%;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  margin: 11.5px 0;
  border-color: transparent;
  color: #fafafa;
  background-color: #bdbdbd;
}

.emotion-5 {
  width: 2px;
  background-color: #bdbdbd;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-6 {
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 6px 16px;
  text-align: right;
}

<div>
  <ul
    class="MuiTimeline-root MuiTimeline-positionLeft emotion-0"
  >
    <li
      class="MuiTimelineItem-root MuiTimelineItem-positionLeft MuiTimelineItem-missingOppositeContent emotion-1"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionLeft emotion-2"
      >
        Opposite
      </div>
      <div
        class="MuiTimelineSeparator-root emotion-3"
      >
        <span
          class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey emotion-4"
        />
        <span
          class="MuiTimelineConnector-root emotion-5"
        />
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionLeft emotion-6"
      >
        Test
      </div>
    </li>
    <li
      class="MuiTimelineItem-root MuiTimelineItem-positionLeft MuiTimelineItem-missingOppositeContent emotion-1"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionLeft emotion-2"
      >
        Opposite
      </div>
      <div
        class="MuiTimelineSeparator-root emotion-3"
      >
        <span
          class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey emotion-4"
        />
        <span
          class="MuiTimelineConnector-root emotion-5"
        />
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionLeft emotion-6"
      >
        Test
      </div>
    </li>
    <li
      class="MuiTimelineItem-root MuiTimelineItem-positionLeft MuiTimelineItem-missingOppositeContent emotion-1"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionLeft emotion-2"
      >
        Opposite
      </div>
      <div
        class="MuiTimelineSeparator-root emotion-3"
      >
        <span
          class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey emotion-4"
        />
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionLeft emotion-6"
      >
        Test
      </div>
    </li>
  </ul>
</div>
`;
